### PR TITLE
[advanced digitizing] Additional tweaks to the 2-circle intersection tools

### PR DESCRIPTION
**Pull request by nirvn**
## Description

@uclaros , PR fixes the ESC and right-click handling, as well as adjusted labels as per your suggestion. 

Screenshot from 2024-08-01 14-05-22

### CODE REVIEW

File: src/gui/qgsadvanceddigitizingdockwidget.cpp [hunk 1]
```diff
@@ -1698,6 +1698,11 @@ bool QgsAdvancedDigitizingDockWidget::canvasKeyPressEventFilter( QKeyEvent *e )
 
 void QgsAdvancedDigitizingDockWidget::clear()
 {
+  if ( mCurrentTool )
+  {
+    mCurrentTool->deleteLater();
+  }
+
```
Comment (uclaros) [inline src/gui/qgsadvanceddigitizingdockwidget.cpp:1705]: I'm afraid this is not enough to handle the right clicking case.
We should be capturing the right click in `QgsAdvancedDigitizingCirclesIntersectionTool` and consume it so it never ends at the actual mapTool.
Consider the following example:
- User digitizes a line, has already added two points
- selects 2 circle intersection
- if he right clicks, his line is finalized, while he should actually end up just like having pressed `Esc`: continue digitizing

Comment (nirvn) [reply src/gui/qgsadvanceddigitizingdockwidget.cpp:1705]: @uclaros , right I wasn't quite sure what would be the ideal option here. You think it'd be better for it not to act as a finalization action when this tool is active?

Comment (nirvn) [reply src/gui/qgsadvanceddigitizingdockwidget.cpp:1705]: For 2 circle intersection, probably not. I'll have the abstract tool able to either take or pass event, and for this intersection tool I'll take (and cancel the cogo operation as suggested here).

Comment (nirvn) [reply src/gui/qgsadvanceddigitizingdockwidget.cpp:1721]: @uclaros , done.

File: src/gui/qgsadvanceddigitizingtools.cpp [hunk 2]
```diff
@@ -297,3 +316,23 @@ void QgsAdvancedDigitizingCirclesIntersectionTool::paint( QPainter *painter )
...
+  if ( event->type() == QEvent::ShortcutOverride || event->type() == QEvent::KeyPress )
+  {
+    if ( QKeyEvent *keyEvent = dynamic_cast<QKeyEvent *>( event ) )
+    {
+      if ( keyEvent->key() == Qt::Key_Escape )
+      {
+        if ( mToolWidget )
+        {
+          mToolWidget->deleteLater();
+        }
```
Comment (uclaros) [inline src/gui/qgsadvanceddigitizingtools.cpp:331]: Is this actually needed, since it's handled in the destructor?

Comment (nirvn) [reply src/gui/qgsadvanceddigitizingtools.cpp]: @uclaros , not anymore no, removed.

**Review by uclaros — COMMENTED** [review]
Thanks for bearing with me, it's almost there!!

Three last bits:
- When cancelling, the canvas items are only updated on mouse move. Could you add a paint request to clear them out at once?
- In `QgsAdvancedDigitizingCirclesIntersectionTool::canvasMoveEvent`, can you add something like
  ```
    event->setMapPoint( mP1Closest ? mP1 : mP2 );
    event->setAccepted( true );
    return;
  ```

  when we have a valid `mP1`? It would be really useful to pass that last point to the actual map tool and update the rubberband.
- I'm not sure there is a gain in using `QgsDoubleSpinbox` for the coordinates widgets. If you insist in keeping them though, can you explicitly set their clear value?

**Comment by nirvn** [comment]
@uclaros , QgsDoubleSpinbox gives us the possibility to do "10+10" for e.g., so we'll keep for coordinates, can come in handy.

**Comment by nyalldawson** [comment]
@uclaros Qgis...SpinBox also allows ctrl+scroll for fine adjustment. For these reasons we should ALWAYS use the Qgis versions, never the base qt ones

**Comment by uclaros** [comment]
@nyalldawson I'm not suggesting to use the base qt ones, I'm questioning the need for a spinbox at all! I can't think of a place we use spinboxes for coordinates in QGIS.

**Comment by nirvn** [comment]
@uclaros , last bits in.

**Review by uclaros — APPROVED** [review]
Thanks for taking care of those issues!

**Pull request merged**